EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -2071,4 +2073,20 @@ void V2::World::updateCountryDetails()
 			}
 		}
 	}
+}
+
+std::optional<std::string> V2::World::getShatteredHreTag()
+{
+	for (const auto& mod: theConfiguration.getMods())
+	{
+		const auto& modFiles = commonItems::GetAllFilesInFolder(mod.path);
+		for (const auto& file: modFiles)
+		{
+			if (file.substr(0, 8) == "i_am_hre")
```

Comment:
Please rework this so it opens i_am_hre.txt and reads the tag from inside (first 3 letters). It'll look cleaner. Also check if you're actually getting 3 characters and return nullopt for failure.

Reply:
Does any of the other converters do something like this, so that I can copy it?

Reply:
They all do, that's how they read uncompressed savegames:
1. open a file if it exists, and pipe it into a string. see EU4::World() when it does that with entire save for example.
2. check if your string is at minimum 3 characters and that *first* 3 are alphanumeric. Drop excess.
3. see if there's a tag registered for that eu4 tag. You can just ask countryMapper with .getV2Tag() to get you something.
4. Go on with your business.